### PR TITLE
Ylva and Matildas Auth-project

### PR DESCRIPTION
**Pull request by sssylvasss**
Link to my deployed project:
https://beepage.netlify.app/

it still don't work as it should. We have a 1 on 1 on Monday to try to fix it.

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,4 +10,4 @@ Describe how you approached to problem, and what tools and techniques you used t
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://beepage.netlify.app/
```
Comment (KarolCislowski) [inline README.md:13]: To fix it you need to create `_redirects` file in the `public` directory of your react project and paste it inside:
``` 
/* /index.html 200
```


image

File: backend/server.js [hunk 2]
```diff
@@ -1,30 +1,122 @@
...
+    unique: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
```
Comment (KarolCislowski) [inline backend/server.js:28]: Nice model 👍🏻

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,9 +1,37 @@
 import React from 'react'
+// import styled from 'styled-components'
```
Comment (KarolCislowski) [inline frontend/src/App.js:2]: 🧹

File: frontend/src/App.js [hunk 4]
```diff
@@ -1,9 +1,37 @@
...
+import { BrowserRouter, Switch, Route } from 'react-router-dom'
+import { Provider } from 'react-redux'
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+
+import { thoughts } from './reducers/thoughts'
+import { user } from './reducers/user'
+
+import {Thoughts} from './components/Thoughts'
+import {RegistrationForm} from './components/RegistrationForm'
+// import {Start} from './components/Start'
```
Comment (KarolCislowski) [inline frontend/src/App.js:13]: 🧹

File: frontend/src/components/RegistrationForm.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    const accessToken = useSelector(store => store.user.accessToken);
+    const dispatch = useDispatch();
+    const history = useHistory();
+
+    useEffect(() => {
+    
+        if (accessToken) {
+            history.push('/thoughts');
+        }
+    }, [accessToken, history]);
```
Comment (KarolCislowski) [inline frontend/src/components/RegistrationForm.js:23]: indentation should be set to 2 spaces.

File: frontend/src/components/RegistrationForm.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            .then(data => {
+           
+                if (data.success) {
+                    batch(() => {
+                        dispatch(user.actions.setUsername(data.username));
+                        dispatch(user.actions.setAccessToken(data.accessToken));
+                        dispatch(user.actions.setErrors(null));
+                    });
+                } else {
+                    console.log('nej')
```
Comment (KarolCislowski) [inline frontend/src/components/RegistrationForm.js:46]: 🧹 console.log 🧹

File: frontend/src/components/Start.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+
+
+// export const Start = () => {
+//     return(
+//         <>
+//     <Link to ='/registration'> Are You Ready to Talk?</Link>
+//         </>
+//     )
+// }
```
Comment (KarolCislowski) [inline frontend/src/components/Start.js:12]: why did you comment-out a whole component? You just need to unmount it from the other components.

File: frontend/src/reducers/user.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        },
+        setLogOut: () => {
+            return {
+                username: null,
+                accessToken: null,
+                errors: null
+            };
+        },   
+    },
+});
```
Comment (KarolCislowski) [inline frontend/src/reducers/user.js:28]: You need to store your data in the local storage to avoid logging out your user after refreshing the page.

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
It is really nice app but you need to clean it up and persist your user in the local storage to avoid refreshing.

Because your graduation is so soon please let me know on slack (a short note with a link to this PR) when you finish fixing your app.

**Review by jesinkey — COMMENTED** [review]
Not anything to add to Karols comments. Glad too see its working after all the trouble you had!

File: README.md (hunk 1 above)
Comment (sssylvasss) [reply README.md:13]: We already have that is our public folder?

File: frontend/src/components/Start.js (hunk 7 above)
Comment (sssylvasss) [reply frontend/src/components/Start.js]: Just wanted to make it easy to see (for ourself) that we did not use it.

**Review by KarolCislowski — APPROVED** [review]
Great App 🥳

**Pull request closed without merging**